src/main/bash/sdkman-completion.sh

```diff
@@ -0,0 +1,78 @@
+#!/usr/bin/env bash
+
+#
+#   Copyright 2017 Marco Vermeulen
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+
+function __sdk_completion() {
+	local -r shell="$1"
+
+	case "$shell" in
+    	bash)
+    		__sdkman_completion_bash
+			;;
+		zsh)
+			__sdkman_completion_zsh
+			;;
+		*)
+			__sdkman_echo_red "No shell provided." && return 1
+	esac
+}
+
+function __sdkman_completion_bash() {
+	\cat <<- 'EOF'
+	_sdk_completion() {
```

Comment:
How about adding the content of this file to a new folder under `~/.sdkman` called something like `contrib`? That way we can version this file and have better control over it.

Reply:
Good point, here documents are also hard to indent. If anyone gonna make modifications to it, he'll probably mess up the indentation. That anyone would probably be me 😂.

Reply:
I suppose this needs to be done in the `sdkman-hooks` repository?

Reply:
Should I also create something like `ext.installContribDir` in `build.gradle`?